server/src/main/java/org/elasticsearch/rest/BaseRestHandler.java

```diff
@@ -128,7 +128,15 @@ public final void handleRequest(RestRequest request, RestChannel channel, NodeCl
                 request.contentStream().setHandler(new HttpBody.ChunkHandler() {
                     @Override
                     public void onNext(ReleasableBytesReference chunk, boolean isLast) {
-                        chunkConsumer.handleChunk(channel, chunk, isLast);
+                        try {
+                            chunkConsumer.handleChunk(channel, chunk, isLast);
+                        } catch (Exception e) {
+                            // Release the chunk if the handler fails before releasing it in exceptional cases
+                            if (chunk.hasReferences()) {
+                                chunk.decRef();
+                            }
+                            throw e;
```

Comment:
Personally I feel it is safter to have some exception handling here just in case any handler misbehaves. That said, it is not an issue in practice since there is only one such handler. I have removed it as you suggested and also added javadoc to the `handleChunk` method to state that handler must close the chunk so that the contract it explicit.